clang/src/Clang/HighLevel/Documentation.hs

```diff
@@ -34,57 +35,63 @@ import GHC.Generics (Generic)
 -- The @CXComment_Null@ kind is not represented by this type.
 -- @'Maybe' 'Comment'@ is used instead, where a @CXComment_Null@ comment is
 -- represented by 'Nothing'.
-data Comment = Comment {
+--
+-- The 'ref' type parameter is to be filled on the hs-bindgen
+-- so that we can have a type that is able to go through all the name mangling
+-- passes. This data type will allow one to cross reference C identifiers when
+-- translating from Doxygen to Haddocks.
+--
+data Comment ref = Comment {
       -- | Clang display name of the cursor
       --
       -- This is included so the original C name can be included in the
       -- translated documentation.
       commentCName :: Text
 
       -- | Children of a the comment
-    , commentChildren :: [CommentBlockContent]
+    , commentChildren :: [CommentBlockContent ref]
     }
-  deriving stock (Show, Eq, Generic)
+  deriving stock (Functor, Show, Eq, Generic)
 
 -- | Reified Clang comment block content
-data CommentBlockContent =
+data CommentBlockContent ref =
       Paragraph {
-        paragraphContent :: [CommentInlineContent]
+        paragraphContent :: [CommentInlineContent ref]
       }
     | BlockCommand {
         blockCommandName :: Text
       , blockCommandArgs :: [Text]
-      , blockCommandParagraph :: [CommentInlineContent]
+      , blockCommandParagraph :: [CommentInlineContent ref]
       }
     | ParamCommand {
         paramCommandName                :: Text
       , paramCommandIndex               :: Maybe Int
       , paramCommandDirection           :: Maybe CXCommentParamPassDirection
       , paramCommandIsDirectionExplicit :: Bool
-      , paramCommandContent             :: [CommentBlockContent]
+      , paramCommandContent             :: [CommentBlockContent ref]
       }
     | TParamCommand {
         tParamCommandName     :: Text
       , tParamCommandPosition :: Maybe [(Int, Int)]
-      , tParamCommandContent  :: [CommentBlockContent]
+      , tParamCommandContent  :: [CommentBlockContent ref]
       }
     | VerbatimBlockCommand {
         verbatimBlockLines :: [Text]
       }
     | VerbatimLine {
         verbatimLine :: Text
       }
-  deriving stock (Show, Eq, Generic)
+  deriving stock (Functor, Show, Eq, Generic)
 
 -- | Reified Clang comment inline content
-data CommentInlineContent =
+data CommentInlineContent ref =
       TextContent {
         textContent :: Text
       }
     | InlineCommand {
         inlineCommandName       :: Text
       , inlineCommandRenderKind :: CXCommentInlineCommandRenderKind
-      , inlineCommandArgs       :: [Text]
+      , inlineCommandArgs       :: [Either Text ref]
```

Comment:
Add a new constructor to `CommentInlineContent` for `InlineRefCommand` and still sanitize the command arguments but punctuation marks should be in a `TextContent`.